aptos-move/framework/aptos-framework/sources/governed_gas_pool.move

```diff
@@ -487,4 +609,38 @@ module aptos_framework::governed_gas_pool {
         coin::deposit(@0xdddd, withdraw);
     }
 
+    #[test(aptos_framework = @aptos_framework, user = @0x1111)]
+    /// Verify low-balance event emits when crossing below dynamic threshold and resets above
+    fun test_low_balance_event_flow(aptos_framework: &signer, user: &signer) acquires GovernedGasPool, GovernedGasPoolExtension, AptosCoinMintCapability, ThresholdParams {
+        // initialize framework + GGP + coin
+        initialize_for_test(aptos_framework);
+
+        // create user and register
+        aptos_account::create_account(signer::address_of(user));
+        aptos_account::register_apt(user);
+
+        // set dynamic threshold: 1000 octas
+        set_threshold_params(aptos_framework, 1000, 1, 0, 1);
+
+        // fund pool above threshold
+        mint_for_test(governed_gas_pool_address(), 2000);
+
+        // ensure flag is false initially
+        let ext0 = borrow_global<GovernedGasPoolExtension>(@aptos_framework);
+        assert!(!ext0.low_event_emitted, 1);
+
+        // withdraw 1500 -> pool = 500 < 1000 => should emit event and set flag
+        let c = withdraw_staking_reward<AptosCoin>(1500);
+        let ext1 = borrow_global<GovernedGasPoolExtension>(@aptos_framework);
+        assert!(ext1.low_event_emitted, 2);
```

Comment:
First, you should test a withdrawal with a threshold above to test the normal process. We must test that we don't generate a false event.